### DISCUSSION TITLE
SX126x Output Power - OCP Config

### DISCUSSION THREAD

**Discussion by mgergos** [jgromes/RadioLib#829] [General]
When using SX1262, SX1268, or STM32WLx at 22 dBm, I note the library sets OCP Config to 60 mA.
Should this be set to 140 mA?
What am I missing?

**Comment by jgromes** [comment]
Correct, 60 mA is the default value for SX126x - why do you think it should be set higher?

**Reply by mgergos** [reply]
Hi Jan,

a) My measured RF output power is 21.5 dBm when OCP is set to 140 mA, but only 16-17 dBm when OCP is set to 60 mA.
b) Page 32 of datasheet (DS_SX1261-2_V2) indicates an OCP value of 0x38 (140 mA) for SX1262.
c) Page 17 of datasheet (DS_SX1261-2_V2) specifies current consumptions of >100 mA at 22 dBm (most of which would be PA).

In the interim, I can supplement my radio.begin() function with radio.setCurrentLimit(140);
However, ideally setOutputPower() would update the OCP based upon the power level.

Curiously, your setOutputPower() reads and writes the OCP value without changing it.
Maybe this is something you were planning but slipped you mind? Or maybe I'm completely off my chops.

Cheers,

**Reply by jgromes** [reply]
Ah, OK - the issue is not actually the default OCP value, since the default output power value is 10 dBm. Automated OCP configuration based on the output power could be added, if it's linear enough. I chose to use 60 mA as the default just for consistency accross SX126x modules.

> Curiously, your setOutputPower() reads and writes the OCP value without changing it.

See page 32 of the datasheet (DS.SX1261-2.W.APP Rev 2.1):

> The OCP is configurable by steps of 2.5 mA and the default value is re-configured automatically each time the function
SetPaConfig(...) is called. If the user wants to adjust the OCP value, it is necessary to change the register as a second step after
calling the function SetPaConfig(...).

In RadioLib, user could have changed the OCP configuration before calling `setOutputPower()`, so the OCP value is cached and then restored.
